### PR TITLE
feat: avoid searching for suggested alternative artworks on every page (DIA-895)

### PR DESCRIPTION
**Pull request by joeyAghion**
This aims to defer a relatively expensive search operation (using the `filter/artworks` API) that's currently happening on every artwork page.

The recent experiment to prefetch upon viewport entry (https://github.com/artsy/force/pull/14520) resulted in _many_ more `artworkRoutes_ArtworkQuery` queries (as visitors scroll through artwork grids). It currently depends on multiple search operations, which cascaded into widespread latency. This PR eliminates a count of similar artworks, which is only used to decide whether to reveal a tooltip and "show more" button in the case of post-auction artworks. The actual similar artworks are already deferred to the client.

Screenshot 2024-09-24 at 4 12 42 PM

In this PR, the tooltip and button are relocated into the already-deferred component. That eliminates a search operation completely in most cases, or in the case of post-auction lots, reuses data already queried by the client.

### CODE REVIEW

File: src/Apps/Artwork/Components/ArtworkAuctionCreateAlertHeader/ArtworkAuctionCreateAlertHeader.tsx [hunk 1]
```diff
@@ -218,11 +195,6 @@ export const ArtworkAuctionCreateAlertHeaderFragmentContainer = createFragmentCo
             name
           }
         }
-        savedSearch {
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/ArtworkAuctionCreateAlertHeader/ArtworkAuctionCreateAlertHeader.tsx:221]: oh yea!

File: src/Apps/Artwork/Components/ArtworkAuctionCreateAlertHeader/ArtworkAuctionCreateAlertHeader.tsx [hunk 2]
```diff
@@ -161,24 +155,7 @@ const ArtworkAuctionCreateAlertHeader: FC<ArtworkAuctionCreateAlertHeaderProps>
...
-            </Column>
-
-            {!!displaySeeMoreButton && (
-              <Column span={2} start={6}>
-                <SuggestedArtworksButton />
-              </Column>
-            )}
-          </>
-        )}
+        <SuggestedArtworksShelfQueryRenderer {...criteria} />
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/ArtworkAuctionCreateAlertHeader/ArtworkAuctionCreateAlertHeader.tsx:158]: yup

**Review by mzikherman — APPROVED** [review]
Excellent!

**Review by damassi — COMMENTED** [review]
Nice @joeyAghion, yr getting it 💯

**Pull request merged**